From d9a228a3f8e4278588afec11ba104ba531c4a014 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Tue, 27 Jul 2021 12:18:30 +0800
Subject: [PATCH 20/45] [FIO fromtree] LF-4200 imx8m: soc: drop phy-reset-gpios
 for fec

Need to drop phy-reset-gpios before booting linux, this property
is legacy property and replaced with reset-gpios.

If provide both, kernel would failed to request the same gpio twice
and cause fec not work.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Michael Scott <mike@foundries.io>
---
 arch/arm/mach-imx/imx8m/soc.c | 22 ++++++++++++----------
 1 file changed, 12 insertions(+), 10 deletions(-)

diff --git a/arch/arm/mach-imx/imx8m/soc.c b/arch/arm/mach-imx/imx8m/soc.c
index c9e0c414bd..7b02376a72 100644
--- a/arch/arm/mach-imx/imx8m/soc.c
+++ b/arch/arm/mach-imx/imx8m/soc.c
@@ -1072,28 +1072,30 @@ static int disable_cpu_nodes(void *blob, u32 disabled_cores)
 #if defined(CONFIG_IMX8MM)
 static int cleanup_nodes_for_efi(void *blob)
 {
-	static const char * const usbotg_path[] = {
-		"/soc@0/bus@32c00000/usb@32e40000",
-		"/soc@0/bus@32c00000/usb@32e50000"
-		};
+	static const char * const path[][2] = {
+		{ "/soc@0/bus@32c00000/usb@32e40000", "extcon" },
+		{ "/soc@0/bus@32c00000/usb@32e50000", "extcon" },
+		{ "/soc@0/bus@30800000/ethernet@30be0000", "phy-reset-gpios" },
+		{ "/soc@0/bus@30800000/ethernet@30bf0000", "phy-reset-gpios" }
+	};
 	int nodeoff, i, rc;
 
-	for (i = 0; i < ARRAY_SIZE(usbotg_path); i++) {
-		nodeoff = fdt_path_offset(blob, usbotg_path[i]);
+	for (i = 0; i < ARRAY_SIZE(path); i++) {
+		nodeoff = fdt_path_offset(blob, path[i][0]);
 		if (nodeoff < 0)
 			continue; /* Not found, skip it */
-		debug("Found %s node\n", usbotg_path[i]);
+		debug("Found %s node\n", path[i][0]);
 
-		rc = fdt_delprop(blob, nodeoff, "extcon");
+		rc = fdt_delprop(blob, nodeoff, path[i][1]);
 		if (rc == -FDT_ERR_NOTFOUND)
 			continue;
 		if (rc) {
 			printf("Unable to update property %s:%s, err=%s\n",
-			       usbotg_path[i], "extcon", fdt_strerror(rc));
+			       path[i][0], path[i][1], fdt_strerror(rc));
 			return rc;
 		}
 
-		printf("Remove %s:%s\n", usbotg_path[i], "extcon");
+		printf("Remove %s:%s\n", path[i][0], path[i][1]);
 	}
 
 	return 0;
-- 
2.35.1

